extensions/shared/src/main/java/app/revanced/extension/shared/patches/spoof/requests/StreamingDataRequest.java

```diff
@@ -75,11 +71,15 @@ protected boolean removeEldestEntry(Entry eldest) {
                 }
             });
 
-    public static String getLastSpoofedClientName() {
-        return lastSpoofedClientType == null
-                ? "Unknown"
-                : lastSpoofedClientType.getFriendlyName();
-    }
+    private static final Map<String, ClientType> clientTypeMap = Collections.synchronizedMap(
```

Comment:
Still using two HashMaps, so it's not good in terms of performance

Reply:
The performance different is not noticeable at all. For more information, before this PR, you used streamingData as map key. It caused a significant delay, since the map have to hash the whole byte array. Use videoId as a key fix this